packages/firestore/src/local/local_store.ts

```diff
@@ -1057,152 +1048,137 @@ export function newLocalStore(
+    IndexedDbPersistence // We only support IndexedDb in multi-tab mode.
+  );
+  return persistenceImpl.getActiveClients();
+}
 
-  lookupMutationDocuments(batchId: BatchId): Promise<MaybeDocumentMap | null> {
-    return this.persistence.runTransaction(
-      'Lookup mutation documents',
+// PORTING NOTE: Multi-Tab only.
+export function getCachedTarget(
+  localStore: LocalStore,
+  targetId: TargetId
+): Promise<Target | null> {
+  const localStoreImpl = debugCast(localStore, LocalStoreImpl);
+  const targetCacheImpl = debugCast(
+    localStoreImpl.targetCache,
+    IndexedDbTargetCache // We only support IndexedDb in multi-tab mode.
+  );
+  const cachedTargetData = localStoreImpl.targetDataByTarget.get(targetId);
+  if (cachedTargetData) {
+    return Promise.resolve(cachedTargetData.target);
+  } else {
+    return localStoreImpl.persistence.runTransaction(
+      'Get target data',
       'readonly',
       txn => {
-        return this.mutationQueue
-          .lookupMutationKeys(txn, batchId)
-          .next(keys => {
-            if (keys) {
-              return this.localDocuments.getDocuments(
-                txn,
-                keys
-              ) as PersistencePromise<MaybeDocumentMap | null>;
-            } else {
-              return PersistencePromise.resolve<MaybeDocumentMap | null>(null);
-            }
-          });
+        return targetCacheImpl
+          .getTargetDataForTarget(txn, targetId)
+          .next(targetData => (targetData ? targetData.target : null));
       }
     );
   }
+}
 
-  removeCachedMutationBatchMetadata(batchId: BatchId): void {
-    this.mutationQueue.removeCachedMutationKeys(batchId);
-  }
-
-  setNetworkEnabled(networkEnabled: boolean): void {
-    this.persistence.setNetworkEnabled(networkEnabled);
-  }
-
-  getActiveClients(): Promise<ClientId[]> {
-    return this.persistence.getActiveClients();
-  }
-
-  getTarget(targetId: TargetId): Promise<Target | null> {
-    const cachedTargetData = this.targetDataByTarget.get(targetId);
-
-    if (cachedTargetData) {
-      return Promise.resolve(cachedTargetData.target);
-    } else {
-      return this.persistence.runTransaction(
-        'Get target data',
-        'readonly',
-        txn => {
-          return this.targetCache
-            .getTargetDataForTarget(txn, targetId)
-            .next(targetData => (targetData ? targetData.target : null));
-        }
-      );
-    }
-  }
-
-  getNewDocumentChanges(): Promise<MaybeDocumentMap> {
-    return this.persistence
-      .runTransaction('Get new document changes', 'readonly', txn =>
-        this.remoteDocuments.getNewDocumentChanges(
-          txn,
-          this.lastDocumentChangeReadTime
-        )
+/**
+ * Returns the set of documents that have been updated since the last call.
+ * If this is the first call, returns the set of changes since client
+ * initialization. Further invocations will return document changes since
+ * the point of rejection.
```

Comment:
Can you clarify what is meant by `rejection`?

Reply:
Changed to `will return document that have changed since the prior call`.